Hardware/Hardware/CardReader/CardReaderService.swift

```diff
@@ -26,7 +26,7 @@ public protocol CardReaderService {
 
     /// Starts the service.
     /// That could imply, for example, that the reader discovery process starts
-    func start()
+    func start(_ adapter: CardReaderNetworkingAdapter)
```

Comment:
I appreciate it is following the adapter pattern, but I wonder if tokenProvider might be a better argument name

Reply:
lol. Yes, I think provider would be better https://github.com/woocommerce/woocommerce-ios/pull/3772#discussion_r586361003